### ISSUE TITLE
1.16.5: Any way to have a Rule respond to entire channel?

### ISSUE THREAD

**Issue by mav2020** [kangarko/ChatControl#996]
I have been trying to figure this out, but can't find anything in the wiki or files to help, so I am hoping I am just missing something. Is there a way to have a rule show to everyone in the channel from which it was sent? For example, when using:

match ^how (do|can|to)(| I| you) (set|create|place)(| a) home
then warn sethome

only the sender will see it. But I want everyone in the channel to see the answer that is in 'sethome.yml'. I thought of using 'notify permission', but then if people leave that chat for another one, but have the permission to join the chat, they will still see the response. Is there a way to accomplish this that I am missing? Thanks.

Priority: Low
Minecraft Version: 1.16.5
MySQL: true
BungeeCord: false

**Comment by ElBananaa** [comment]
> I have been trying to figure this out, but can't find anything in the wiki or files to help, so I am hoping I am just missing something. Is there a way to have a rule show to everyone in the channel from which it was sent? For example, when using:
> 
> match ^how (do|can|to)(| I| you) (set|create|place)(| a) home
> then warn sethome
> 
> only the sender will see it. But I want everyone in the channel to see the answer that is in 'sethome.yml'. I thought of using 'notify permission', but then if people leave that chat for another one, but have the permission to join the chat, they will still see the response. Is there a way to accomplish this that I am missing? Thanks.
> 
> Priority: Low
> Minecraft Version: 1.16.5
> MySQL: true
> BungeeCord: false

Hey!
To do so, I think you could use the `require script` variable : 
`require script '{player_channel_mode_{channel}}' == 'read'` (Don't forget to replace `{channel` with the actual channel. (https://github.com/kangarko/ChatControl-Red/wiki/Rules#require-script-javascript-returning-a-boolean)

**Comment by mav2020** [comment]
I tried a few variations of that and didn't get the results I wanted. It didn't match unless I used 'write' which makes sense. But more importantly it either doesn't show to others if I use 'then warn' or shows to all players with the permission if I use below. Did I do it wrong?

match ^how (do|can|to)(| I| you) (set|create|place)(| a) home
require script '{player_channel_mode_standard}' == 'write'
then notify chatcontrol.channel.autojoin.standard.write sethome

on top of that, `then notify` seems to go above the original message which is less useful if I don't deny the message (warn goes below it like I would like). Is there a way to remedy that with some kind of `wait 1 tick` type thing (or should that be a new question)?

Basically, I want to send a message to the sender's channel that can be seen by all who are in there (or at least those that are set to write in there). Warn works great, but then only the sender sees it. Thanks for your help.
